### DISCUSSION TITLE
How to Export Data Model File to Feed Into ChatGPT

### DISCUSSION THREAD

**Discussion by juaguirre** [TabularEditor/TabularEditor3#1013] [🙋♀️TE3 Q&A]
I am having great, albeit imperfect, results using ChatGPT to write DAX, however I find myself spending a lot of time describing the model tables, fields and relationships, how can I export the data model to a file so I can feed into ChatGPT and save lots of time?

**Comment by otykier** [comment]
Have you tried loading the raw model JSON into ChatGPT? If that doesn’t work, you can always use C# scripting to export the model metadata in a suitable format, but you will probably have to experiment a bit to figure out what works best.

There are also these resources which may help or serve as inspiration:

- [Automatically generating measure descriptions for Power BI and Analysis Services with ChatGPT and Tabular Editor](https://darren.gosbell.com/2023/02/automatically-generating-measure-descriptions-for-power-bi-and-analysis-services-with-chatgpt-and-tabular-editor/)
- [Translate your measures using Tabular Editor and GPT3](https://blog.tabulareditor.com/2023/03/24/translate-your-measures-using-tabular-editor-and-gpt3/)

**Comment by mlonsk** [comment]
Hi @juaguirre 

I remember seeing [this post](https://www.linkedin.com/posts/brianjuliusdc_write-and-optimize-dax-at-a-world-class-level-activity-7113883464650645504-Vjb4?utm_source=share&utm_medium=member_desktop) from Brian Julius about a process for uploading the .bim file directly to ChatGPT
Maybe an source for inspiration for you?
